scripts/test.sh

```diff
@@ -1,14 +1,29 @@
 #!/bin/bash
 set -ex
 
-cp -r fixtures/2.5/ fixtures/2.6
-for file in `ls fixtures/*_expected.rb` `ls fixtures/$(ruby -v | grep -o '\d\.\d')/*_expected.rb`
-do
-    time ruby --disable=gems src/rubyfmt.rb `echo $file | sed s/expected/actual/` > /tmp/out.rb
-    diff -u /tmp/out.rb $file
-    if [[ $? -ne 0 ]]
-    then
-        echo "got diff"
-        exit 1
-    fi
-done
+test_folder() {
+    for file in `ls $1/*_expected.rb`
+    do
+        time ruby --disable=gems src/rubyfmt.rb `echo $file | sed s/expected/actual/` > /tmp/out.rb
+        diff /tmp/out.rb $file
+        if [[ $? -ne 0 ]]
+        then
+            echo "got diff"
+            exit 1
+        fi
+    done
+}
+
+test_folder fixtures/
+
+RUBY_VERSION=$(ruby -v | grep -o "[0-9].[0-9]" | head -n 1)
+echo $RUBY_VERSION
+if [[ `echo "2.5<=$RUBY_VERSION" | bc -l` -ne 0 ]]
+then
+    test_folder fixtures/2.5
+fi
+
+if [[ `echo "2.6<=$RUBY_VERSION" | bc -l` -ne 0 ]]
+then
+    test_folder fixtures/2.6
+fi
```

Comment:
What do you think about making this automatically discover directories for new versions? Would probably make it easier for contributors to add version-specific tests.

This works for me on macOS:

```shell
for dir in $(find fixtures -type d -name '2.*')
do
    fixture_version=$(basename "$dir")
    if [[ $(echo "$fixture_version<=$RUBY_VERSION" | bc -l) -ne 0 ]]
    then
        test_folder $dir
    fi
done
```

Reply:
Please open a PR for this, right now I'd like to go on a merge spree.